psychopy/sound/backend_ptb.py

```diff
@@ -340,11 +340,13 @@ def __init__(self, value="C", secs=0.5, octave=4, stereo=-1,
     @property
     def isPlaying(self):
         """`True` if the audio playback is ongoing."""
+        self._checkPlaybackFinished()
```

Comment:
Good point, I was thinking we need to check in case it's finished but actually _checkPlaybackFinished doesn't affect that, it's just set to False on `.stop()`

Reply:
Interestingly, because of the change to `isFinished()`, we now have the chain `isFinished()` >> `_checkPlaybackFinished()` >> `_EOS()` >> `.stop()` >> `self._isPlaying = False`...

So we should actually now call `self._checkPlaybackFinished()` before returning `self._isPlaying` like how you originally had it 😅

One more thing that needs to go along with resurrecting this line, change the 
```
        if not self.isPlaying:
            return
```
in `def stop()` to
```
        if not self._isPlaying:
            return
```
Otherwise we will have circular calls that result in infinite loops.

Reply:
I'm not sure it is infinite... You'd get `_checkPlaybackFinished` >> `_EOS` >> `stop` >> `isPlaying` >> `_checkPlaybackFinished`, but on that second `_checkPlaybackFinished` you wouldn't get `_EOS` as `wasFinished` would now be `True`. You also won't get `self._isPlaying = False` unless `isFinished` is `True` and `self._loopsFinished <= self.loops`, which seems reasonable as playing has genuinely stopped. So I think this is fine to leave as is, and certainly testing it out it seems to work.

Reply:
I agree that `wasFinished` stops the infinite loop from happening. But we should call `self._checkPlaybackFinished()` in `isPlaying()` because it now has a possible route to toggle the `_isPlaying` attribute. Imagine I only call `isPlaying()` in a loop to check if a sound has finished playing. Without this call it can't possibly update `_isPlaying`. I'll make a separate PR.